plugins/outputs/loki/loki_test.go

```diff
@@ -0,0 +1,370 @@
+package loki
+
+import (
+	"compress/gzip"
+	"encoding/json"
+	"fmt"
+	"github.com/influxdata/telegraf/testutil"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/stretchr/testify/require"
+)
+
+func getMetric() telegraf.Metric {
+	return testutil.MustMetric(
+		"log",
+		map[string]string{
+			"key1": "value1",
+		},
+		map[string]interface{}{
+			"line":  "my log",
+			"field": 3.14,
+		},
+		time.Unix(123, 0),
+	)
+}
+
+func TestStatusCode(t *testing.T) {
+	ts := httptest.NewServer(http.NotFoundHandler())
+	defer ts.Close()
+
+	u, err := url.Parse(fmt.Sprintf("http://%s", ts.Listener.Addr().String()))
+	require.NoError(t, err)
+
+	tests := []struct {
+		name       string
+		plugin     *Loki
+		statusCode int
+		errFunc    func(t *testing.T, err error)
+	}{
+		{
+			name: "success",
+			plugin: &Loki{
+				Domain: u.String(),
+			},
+			statusCode: http.StatusNoContent,
+			errFunc: func(t *testing.T, err error) {
+				require.NoError(t, err)
+			},
+		},
+		{
+			name: "1xx status is an error",
+			plugin: &Loki{
+				Domain: u.String(),
+			},
+			statusCode: 103,
+			errFunc: func(t *testing.T, err error) {
+				require.Error(t, err)
+			},
+		},
+		{
+			name: "3xx status is an error",
+			plugin: &Loki{
+				Domain: u.String(),
+			},
+			statusCode: http.StatusMultipleChoices,
+			errFunc: func(t *testing.T, err error) {
+				require.Error(t, err)
+			},
+		},
+		{
+			name: "4xx status is an error",
+			plugin: &Loki{
+				Domain: u.String(),
+			},
+			statusCode: http.StatusMultipleChoices,
+			errFunc: func(t *testing.T, err error) {
+				require.Error(t, err)
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ts.Config.Handler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(tt.statusCode)
+			})
+
+			err = tt.plugin.Connect()
+			require.NoError(t, err)
+
+			err = tt.plugin.Write([]telegraf.Metric{getMetric()})
+			tt.errFunc(t, err)
+		})
+	}
+}
+
+func TestContentType(t *testing.T) {
+	ts := httptest.NewServer(http.NotFoundHandler())
+	defer ts.Close()
+
+	u, err := url.Parse(fmt.Sprintf("http://%s", ts.Listener.Addr().String()))
+	require.NoError(t, err)
+
+	tests := []struct {
+		name     string
+		plugin   *Loki
+		expected string
+	}{
+		{
+			name: "default is text plain",
+			plugin: &Loki{
+				Domain: u.String(),
+			},
+			expected: "application/json",
```

Comment:
This test suggests that the default is `json` and not plain-text as the test-name suggests. Am I missing something?